crates/bevy_text/src/text.rs

```diff
@@ -186,12 +186,17 @@ pub enum BreakLineOn {
     /// This is closer to the behavior one might expect from text in a terminal.
     /// However it may lead to words being broken up across linebreaks.
     AnyCharacter,
+    /// Lines will never be broken up.
+    NoWrap,
 }
 
 impl From<BreakLineOn> for glyph_brush_layout::BuiltInLineBreaker {
     fn from(val: BreakLineOn) -> Self {
         match val {
-            BreakLineOn::WordBoundary => glyph_brush_layout::BuiltInLineBreaker::UnicodeLineBreaker,
+            // If `NoWrap` is set the text is given unbounded width, so the `BuiltInLineBreaker` won't be used.
```

Comment:
I think this comment isn't quite accurate, in that I would expect wrapping to still occur if the text contains an explicit newline character. It's just that soft wrapping won't occur at "line-break opportunities".

Reply:
I'm not sure, as the line breaker is only used for soft breaks I think so maybe it's correct?

We need to be very clear that `NoWrap` only applies to soft breaks and not hard breaks caused by a newline though definitely.

Reply:
No I'm wrong, with either enum internally it uses a `xi_unicode::LineBreakIterator` that handles both hard and soft breaks.